src/Pirouette/Monad.hs

```diff
@@ -57,57 +51,40 @@ data PrtError
   | PEOther String
   deriving (Eq, Show)
 
--- | And the actual error raised by the error monad. Make sure to raise these
---  with 'throwError'', so you can get the logger context in your error messages
---  for free.
-data PrtErrorCtx = PrtErrorCtx
-  { logCtx :: [String],
-    message :: PrtError
-  }
-  deriving (Eq, Show)
-
--- | The base monadic layer we use in most places.
-type PirouetteBase m = (MonadError PrtErrorCtx m, MonadLogger m, MonadFail m)
-
--- | 'throwError' variant that automatically gets the logger context.
-throwError' :: (PirouetteBase m) => PrtError -> m a
-throwError' msg = do
-  err <- flip PrtErrorCtx msg <$> context
-  throwError err
+prtError :: PrtError -> a
+prtError = error . show
```

Comment:
What is the reasoning for moving from `MonadFail` to `error`? I'm on the fence here: I like the removal of `MonadError`, but I feel like `error` is too difficult to catch.

Reply:
The reasoning is that we were never catching any of those errors anyway. They were all un-recoverable errors. Things like: "Symbol not defined" is not something we can recover from.